src/templates/ssr/fabric-custom-xl/index.ts

```diff
@@ -0,0 +1,67 @@
+import { post } from '$lib/messenger';
+
+const CACHE_BUST = '%%CACHEBUSTER%%';
+const DapAssetsRoot = `https://s3-eu-west-1.amazonaws.com/adops-assets/dap-fabrics`;
+const DapAssetsFolder: string = '[%DapAssetsFolder%]';
+
+const DapAssetsPath = `${DapAssetsRoot}/${DapAssetsFolder}`;
+const ThirdPartyTag: string = '[%ThirdPartyTag%]';
+const TrackingPixel: string = '[%TrackingPixel%]';
+const ResearchPixel: string = '[%ResearchPixel%]';
+
+const addTrackingPixel = (url: string) => {
+	const pixel = new Image();
+	pixel.src = url + CACHE_BUST;
+};
+
+// relative paths in the CSS need to be replaced with the absolute path
+const replaceAssetLinks = (html: string) => {
+	const re = /url\('\.\/(.*)'\)/g;
+	return html.replace(re, `url('${DapAssetsPath}/$1')`);
+};
+
+const getTag = () => {
+	if (DapAssetsFolder) {
+		return fetch(`${DapAssetsPath}/index.html`)
+			.then((res) => res.text())
+			.then(replaceAssetLinks);
+	} else if (ThirdPartyTag) {
+		return fetch(ThirdPartyTag).then((res) => res.text());
+	}
+	return Promise.reject('No tag found');
+};
+
+/**
+ * This is a hack to insert the html into the DOM,
+ * using innerHtml would not work if the tag contains a script tag
+ * see https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML#security_considerations
+ *
+ * @param tag the html to insert
+ */
+const insertTag = (tag: string) => {
```

Comment:
Some of these functions are copied from `fabric-custom` right? can we move them to a lib and import them to avoid duplication?

Reply:
@Jakeii I've placed the shared logic in the `lib` directory (@lib) for consistency rather than create a new folder in the `SSR` directory - which was causing build issues.